enterprise/backend/test/metabase_enterprise/semantic_search/embedding_test.clj

```diff
@@ -86,3 +92,140 @@
+    (let [embedding1 [1.0 2.0 3.0]
+          embedding2 [-1.0 -2.0 -3.0]
+          embedding3 [0.0 0.5 -0.5]
+          base64-str1 (encode-floats-to-base64 embedding1)
+          base64-str2 (encode-floats-to-base64 embedding2)
+          base64-str3 (encode-floats-to-base64 embedding3)
+          response {:data [{:embedding base64-str1}
+                           {:embedding base64-str2}
+                           {:embedding base64-str3}]}
+          result (#'embedding/extract-base64-response-embeddings response)]
+      (is (= 3 (count result)))
+      (is (float-vectors-approx= embedding1 (nth result 0)))
+      (is (float-vectors-approx= embedding2 (nth result 1)))
+      (is (float-vectors-approx= embedding3 (nth result 2)))))
+
+  (testing "edge cases that return empty results"
+    (testing "handles empty data array"
+      (let [response {:data []}
+            result (#'embedding/extract-base64-response-embeddings response)]
+        (is (= [] result))))
+
+    (testing "handles missing :data key"
+      (let [response {}
+            result (#'embedding/extract-base64-response-embeddings response)]
+        (is (= [] result))))
+
+    (testing "handles zero-length embedding"
+      (let [base64-str (encode-floats-to-base64 [])
+            response {:data [{:embedding base64-str}]}
+            result (#'embedding/extract-base64-response-embeddings response)]
+        (is (= 1 (count result)))
+        (is (= [] (first result))))))
+
+  (testing "handles large embedding vectors"
+    (let [large-embedding (vec (map float (range 1536)))
+          base64-str (encode-floats-to-base64 large-embedding)
+          response {:data [{:embedding base64-str}]}
+          result (#'embedding/extract-base64-response-embeddings response)]
+      (is (= 1 (count result)))
+      (is (= 1536 (count (first result))))
+      (is (float-vectors-approx= (take 5 large-embedding) (take 5 (first result))))
+      (is (float-vectors-approx= (take-last 5 large-embedding) (take-last 5 (first result))))))
+
+  (testing "error cases"
+    (testing "invalid base64 string throws exception"
+      (is (thrown? Exception
+                   (#'embedding/extract-base64-response-embeddings
+                    {:data [{:embedding "not-valid-base64!@#$"}]}))))
+
+    (testing "non-string embedding value throws exception"
+      (is (thrown? Exception
+                   (#'embedding/extract-base64-response-embeddings
+                    {:data [{:embedding 123}]}))))
+
+    (testing "base64 string with invalid byte count (not multiple of 4) throws exception"
+      ;; Create a base64 string that decodes to 3 bytes
+      (let [encoder (Base64/getEncoder)
+            invalid-base64 (.encodeToString encoder (byte-array [1 2 3]))]
+        (is (thrown? Exception
+                     (#'embedding/extract-base64-response-embeddings
+                      {:data [{:embedding invalid-base64}]})))))))
+
+(deftest test-get-embedding
```

Comment:
This just verified that we can call `get-embedding` and `get-embeddings-batched` for each of the known providers.